lib/puppet/type/apache_directive.rb

```diff
@@ -52,4 +56,35 @@ def insync?(is)
   newparam(:target) do
     desc 'The config file to use'
   end
+
+  def self.title_patterns
+    identity = lambda { |x| x }
+    [
+      [
+        /^((\S+)\s+of\s+(.+))$/,
```

Comment:
My ruby is "super bad"!

It looks like I'm doing similar things to what the pg_hba type is doing....

Alas namevar was created so I didn't have to change the other code.  Puppet was treating ':name' as the namevar and when splitting the complex name into attribs the use of the directive name as ':name' was causing duplicate resource definitions.